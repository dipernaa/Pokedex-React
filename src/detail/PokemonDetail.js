import React from 'react';
import Avatar from 'material-ui/Avatar';
import { Card, CardHeader, CardTitle, CardText } from 'material-ui/Card';
import Paper from 'material-ui/Paper';

const styles = {
  avatar: {
    marginLeft: 'auto',
    marginRight: 'auto'
  },
  avatarHolder: {
    alignItems: 'center',
    display: 'flex',
    height: 160,
    marginLeft: 'auto',
    marginRight: 'auto',
    position: 'relative',
    top: 20,
    width: 160,
    zIndex: 3
  },
  card: {
    left: 0,
    margin: 20,
    position: 'absolute',
    right: 0,
    top: 170,
    zIndex: 1
  },
  infoLine: {
    paddingBottom: 5
  },
  statList: {
    margin: 0
  }
};

const PokemonDetail = ({ pokemon }) => (
  pokemon ?
    <div>
      <Paper circle style={styles.avatarHolder}>
        <Avatar size={150} src={pokemon.sprites.front_default} style={styles.avatar}/>
      </Paper>
      <Card style={styles.card}>
        <CardHeader/>
        <CardTitle className="capitalize" title={pokemon.name}/>
        <CardText>
          <div style={styles.infoLine}>
            <strong>Types:</strong>&nbsp;{pokemon.types.map(item => item.type.name).join(', ')}
          </div>
          <div style={styles.infoLine}>
            <strong>Base Experience:</strong>&nbsp;{pokemon.base_experience}
          </div>
          <div style={styles.infoLine}>
            <strong>Height:</strong>&nbsp;{pokemon.height}
          </div>
          <div style={styles.infoLine}>
            <strong>Weight:</strong>&nbsp;{pokemon.weight}
          </div>
          <div style={styles.infoLine}>
            <strong>Stats:</strong>
            <ul style={styles.statList}>
              {pokemon.stats.map((item, index) => <li key={index}>{item.stat.name}: {item.base_stat}</li>)}
            </ul>
          </div>
        </CardText>
      </Card>
    </div>
  : null
);

export default PokemonDetail;
