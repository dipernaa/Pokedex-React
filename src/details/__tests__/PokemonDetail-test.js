import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PokemonDetail from '../PokemonDetail';

const testProps = {
  pokemon: {
    id: '1',
    name: 'pokedude',
    base_experience: 123,
    height: 21,
    weight: 12,
    sprites: {
      front_default: 'http://www.some.url/pic.png'
    },
    stats: [
      {
        base_stat: 42,
        stat: { name: 'hp' }
      },
      {
        base_stat: 12,
        stat: { name: 'speed' }
      }
    ],
    types: [
      {
        type: { name: 'bug' }
      },
      {
        type: { name: 'poison' }
      }
    ]
  }
};

it('renders null when pokemon is not present', () => {
  const wrapper = shallow(
    <PokemonDetail/>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders a PokemonDetail when pokemon is present', () => {
  const wrapper = shallow(
    <PokemonDetail {...testProps}/>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
