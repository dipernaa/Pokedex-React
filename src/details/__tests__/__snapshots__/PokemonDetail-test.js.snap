// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a PokemonDetail when pokemon is present 1`] = `
<div>
  <Paper
    circle={true}
    rounded={true}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 160,
        "marginLeft": "auto",
        "marginRight": "auto",
        "position": "relative",
        "top": 20,
        "width": 160,
        "zIndex": 3,
      }
    }
    transitionEnabled={true}
    zDepth={1}
  >
    <Avatar
      size={150}
      src="http://www.some.url/pic.png"
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
        }
      }
    />
  </Paper>
  <Card
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
    style={
      Object {
        "left": 0,
        "margin": 20,
        "position": "absolute",
        "right": 0,
        "top": 170,
        "zIndex": 1,
      }
    }
  >
    <CardHeader
      avatar={null}
    />
    <CardTitle
      className="capitalize"
      title="pokedude"
    />
    <CardText>
      <div
        style={
          Object {
            "paddingBottom": 5,
          }
        }
      >
        <strong>
          Types:
        </strong>
         
        bug, poison
      </div>
      <div
        style={
          Object {
            "paddingBottom": 5,
          }
        }
      >
        <strong>
          Base Experience:
        </strong>
         
        123
      </div>
      <div
        style={
          Object {
            "paddingBottom": 5,
          }
        }
      >
        <strong>
          Height:
        </strong>
         
        21
      </div>
      <div
        style={
          Object {
            "paddingBottom": 5,
          }
        }
      >
        <strong>
          Weight:
        </strong>
         
        12
      </div>
      <div
        style={
          Object {
            "paddingBottom": 5,
          }
        }
      >
        <strong>
          Stats:
        </strong>
        <ul
          style={
            Object {
              "margin": 0,
            }
          }
        >
          <li>
            hp
            : 
            42
          </li>
          <li>
            speed
            : 
            12
          </li>
        </ul>
      </div>
    </CardText>
  </Card>
</div>
`;

exports[`renders null when pokemon is not present 1`] = `null`;
